src/System.CommandLine/CommandLineConfiguration.cs

```diff
@@ -13,21 +13,21 @@ namespace System.CommandLine
 {
     public class CommandLineConfiguration
     {
-        private IReadOnlyCollection<InvocationMiddleware> _middlewarePipeline;
-        private Func<BindingContext, IHelpBuilder> _helpBuilderFactory;
+        private IReadOnlyCollection<InvocationMiddleware>? _middlewarePipeline;
```

Comment:
nit: Consider moving the defaulting logic into the constructor instead, so the field never has to hold null. Same for `_helpBuilderFactory`